lib/medtronic/medtronicSimulator.js

```diff
@@ -66,6 +77,111 @@ exports.make = function(config){
   }
 
   return {
+    basal: function(event) {
+      ensureTimestamp(event);
+
+      if (currBasal != null) {
+
+        if(currBasal.deliveryType === 'temp' && event.deliveryType === 'temp' && event.duration !== 0) {
+          // temp basal was updated
+          currBasal.with_expectedDuration(currBasal.duration)
+                   .with_duration(Date.parse(event.time) - Date.parse(currBasal.time));
+          event.suppressed = _.clone(currBasal.suppressed);
+        }
+
+        if(currBasal.deliveryType === 'temp' && currPumpSettings) {
+          var currSchedule = currPumpSettings.basalSchedules[currPumpSettings.activeSchedule];
+          var startTime = common.computeMillisInCurrentDay(currBasal);
+          var endTime = startTime + currBasal.duration;
+
+          for(var i in currSchedule) {
+            if((startTime < currSchedule[i].start) && (endTime > currSchedule[i].start)) {
```

Comment:
OK, I've added some logic to handle temp basals crossing midnight with and without schedule changes. The tests pass, but it's not quite as clean as I would like. If there's a better way to do this, please point me in the right direction :)